### DISCUSSION TITLE
Symmetry: parity of the number of fermions

### DISCUSSION THREAD

**Discussion by LGotta** [QuSpin/QuSpin#651] [Q&A] [answered]
Hello, I am writing here to kindly ask you how one would proceed to write a Hamiltonian with the following features:

1.  The quantum particles to be described are fermions with 3 internal degrees of freedom;
2.  the Hamiltonian conserves the parity of the number of fermions $\hat P=\prod_{j,\sigma} e^{i\pi \hat n_{j,\sigma}}$, but not the total number of fermions by itself.

I think in the first point one can proceed as in the Fermi-Hubbard model, but including one spinless fermion Hilbert space copy more in the tensor product. I don't know however how to implement this symmetry defined by the operator $\hat P$, in order to study the spectrum within each symmetry block.
Would you have a draft of a way to do this?
Thanks a lot for your attention.
Sincerely,
Lorenzo

**Comment by mgbukov** [comment, marked as answer]
I guess your best bet is to use the user basis:

https://quspin.github.io/QuSpin/user_basis.html

You can take a 1d lattice and divide the sites into three equally large
sets, each labeling one internal degree of freedom. For the conservation of
parity you can use the next_state function.

See these examples for some inspiration:

https://quspin.github.io/QuSpin/examples/user-basis_example1.html#user-basis-example1-label
https://quspin.github.io/QuSpin/examples/example15.html#example15-label



On Mon, Nov 13, 2023 at 3:09 PM LGotta ***@***.***> wrote:

> Hello, I am writing here to kindly ask you how one would proceed to write
> a Hamiltonian with the following features:
>
>    1. The quantum particles to be described are fermions with 3 internal
>    degrees of freedom;
>    2. the Hamiltonian conserves the parity of the number of fermions $\hat
>    P=\prod_{j,\sigma} e^{i\pi \hat n_{j,\sigma}}$, but not the total
>    number of fermions by itself.
>
> I think in the first point one can proceed as in the Fermi-Hubbard model,
> but including one spinless fermion Hilbert space copy more in the tensor
> product. I don't know however how to implement this symmetry defined by the
> operator $\hat P$, in order to study the spectrum within each symmetry
> block.
> Would you have a draft of a way to do this?
> Thanks a lot for your attention.
> Sincerely,
> Lorenzo
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/QuSpin/QuSpin/discussions/651>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADHSUBXKAFMXSPKANRS77ELYEIZZ7AVCNFSM6AAAAAA7JJVGQCVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZVHA2DOMBQGE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Marin Bukov, PhD
Research Group Leader
Max Planck Institute for the Physics of Complex Systems
Noethnitzer Str. 38
Dresden 01187, Germany

url: https://www.pks.mpg.de/nqd
<http://quantum-dynamics.phys.uni-sofia.bg/>
